File: backend/MotoclubeManager.Api/Models/Membro.cs
namespace MotoclubeManager.Api.Models;

public class Membro
{
    public int Id { get; set; }

    public string Nome { get; set; } = string.Empty;

    public string Apelido { get; set; } = string.Empty;

    public string Telefone { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public DateTime DataIngresso { get; set; }

    public DateTime? DataPromocaoMeioEscudo { get; set; }

    public DateTime? DataPromocaoEscudo { get; set; }

    public DateTime? DataSaida { get; set; }

    public StatusMembro Status { get; set; }

    public CargoMembro Cargo { get; set; }

    public string Observacoes { get; set; } = string.Empty;

    public DateTime DataCriacao { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/membro.ts
export type StatusMembro =
  | "Ativo"
  | "Licenciado"
  | "Suspenso"
  | "Inativo"
  | "Desligado";

export type CargoMembro =
  | "Padrao"
  | "Secretario"
  | "DiretorFinanceiro"
  | "DiretorEventos"
  | "SgtArmas"
  | "VicePresidente"
  | "Presidente";

export interface Membro {
  id: number;
  nome: string;
  apelido: string;
  telefone: string;
  email: string;
  dataIngresso: string;
  dataPromocaoMeioEscudo?: string | null;
  dataPromocaoEscudo?: string | null;
  dataSaida?: string | null;
  status: StatusMembro;
  cargo: CargoMembro;
  observacoes: string;
  dataCriacao: string;
}